exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,2001 @@
+	}
+	if !arg.EndTime.IsZero() && arg.EndTime.Unix() != 0 {
+		params.Set("endTime", strconv.FormatInt((arg.EndTime).UnixMilli(), 10))
+	}
+	path := common.EncodeURLValues(candleStick, params)
+	var resp interface{}
+	err = bi.SendHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		path,
+		spotDefaultRate,
+		&resp)
+	if err != nil {
+		return nil, err
+	}
+	responseData, ok := resp.([]interface{})
+	if !ok {
+		return nil, errUnableToTypeAssertResponseData
+	}
+
+	klineData := make([]CandleStick, len(responseData))
+	for x := range responseData {
+		individualData, ok := responseData[x].([]interface{})
+		if !ok {
+			return nil, errUnableToTypeAssertInvalidData
+		}
+		if len(individualData) != 12 {
+			return nil, errUnexpectedKlineDataLength
+		}
+		var candle CandleStick
+		val, ok := individualData[0].(float64)
+		if !ok {
+			return nil, errUnixMilliSecTypeAssertion
+		}
+		candle.OpenTime = time.UnixMilli(int64(val))
+		if candle.Open, err = convert.FloatFromString(individualData[1]); err != nil {
+			return nil, err
+		}
+		if candle.High, err = convert.FloatFromString(individualData[2]); err != nil {
+			return nil, err
+		}
+		if candle.Low, err = convert.FloatFromString(individualData[3]); err != nil {
+			return nil, err
+		}
+		if candle.Close, err = convert.FloatFromString(individualData[4]); err != nil {
+			return nil, err
+		}
+		if candle.Volume, err = convert.FloatFromString(individualData[5]); err != nil {
+			return nil, err
+		}
+		val, ok = individualData[6].(float64)
+		if !ok {
+			return nil, errUnixMilliSecTypeAssertion
+		}
+		candle.CloseTime = time.UnixMilli(int64(val))
+		if candle.QuoteAssetVolume, err = convert.FloatFromString(individualData[7]); err != nil {
+			return nil, err
+		}
+		if candle.TradeCount, ok = individualData[8].(float64); !ok {
+			return nil, errUnableToTypeAssertTradeCount
+		}
+		if candle.TakerBuyAssetVolume, err = convert.FloatFromString(individualData[9]); err != nil {
+			return nil, err
+		}
+		if candle.TakerBuyQuoteAssetVolume, err = convert.FloatFromString(individualData[10]); err != nil {
+			return nil, err
+		}
+		klineData[x] = candle
+	}
+	return klineData, nil
+}
+
+// GetSinglePriceData to get the latest price for a token symbol or symbols.
+func (bi *Binanceus) GetSinglePriceData(ctx context.Context, symbol currency.Pair) (SymbolPrice, error) {
+	var res SymbolPrice
+	params := url.Values{}
+	symbolValue, err := bi.FormatSymbol(symbol, asset.Spot)
+	if err != nil {
+		return res, err
+	}
+	params.Set("symbol", symbolValue)
+	path := common.EncodeURLValues(tickerPrice, params)
+	return res, bi.SendHTTPRequest(ctx, exchange.RestSpotSupplementary, path, spotDefaultRate, &res)
+}
+
+// GetPriceDatas to get the latest price for symbols.
+func (bi *Binanceus) GetPriceDatas(ctx context.Context) (SymbolPrices, error) {
+	var res SymbolPrices
+	return res, bi.SendHTTPRequest(ctx, exchange.RestSpotSupplementary, tickerPrice, spotDefaultRate, &res)
```

Comment:
Rate limit is 2 when symbol unset: https://docs.binance.us/#get-live-ticker-price